packages/next/next-server/server/incremental-cache.ts

```diff
@@ -125,18 +127,17 @@ export class IncrementalCache {
       }
 
       try {
-        const html = await promises.readFile(
-          this.getSeedPath(pathname, 'html'),
-          'utf8'
-        )
+        const htmlPath = this.getSeedPath(pathname, 'html')
+        const html = await promises.readFile(htmlPath, 'utf8')
+        const { mtime } = await promises.stat(htmlPath)
         const pageData = JSON.parse(
           await promises.readFile(this.getSeedPath(pathname, 'json'), 'utf8')
         )
 
         data = {
           html,
           pageData,
-          revalidateAfter: this.calculateRevalidate(pathname),
+          revalidateAfter: this.calculateRevalidate(pathname, mtime.getTime()),
```

Comment:
Use the `lastModified` file timestamp from the filesystem cache, when refreshing the `LRUCache` content in `IncrementalCache.get`, as this timestamp is when the content was last generated and then written by `IncrementalCache.set` - so is the anchor point from which we should be calculating the `revalidateAfter` timestamp from here.